exporter/elasticexporter/internal/translator/elastic/metadata.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package elastic contains an OTLP exporter for Elastic APM.
+package elastic
+
+import (
+	"fmt"
+
+	otlpresource "github.com/open-telemetry/opentelemetry-proto/gen/go/resource/v1"
+	"go.elastic.co/apm/model"
+	"go.elastic.co/fastjson"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+// EncodeResourceMetadata encodes a metadata line from resource, writing to w.
+func EncodeResourceMetadata(resource *otlpresource.Resource, w *fastjson.Writer) {
+	var agent model.Agent
+	var service model.Service
+	var serviceNode model.ServiceNode
+	var serviceLanguage model.Language
+	var system model.System
+	var k8s model.Kubernetes
+	var k8sPod model.KubernetesPod
+	var labels model.IfaceMap
+
+	for _, attr := range resource.GetAttributes() {
+		switch k := attr.GetKey(); k {
+		case conventions.AttributeServiceName:
+			service.Name = cleanServiceName(attr.GetStringValue())
```

Comment:
Did you mean prefix with namespace?

Reply:
yes that's what I meant, sorry for the confusion

Reply:
I was all set to do that, but I see this comment in the spec:

> Note: service.namespace and service.name are not intended to be concatenated for the purpose of forming a single globally unique name for the service. For example the following 2 sets of attributes actually describe 2 different services (despite the fact that the concatenation would result in the same string):

And looking at other exporters based on the old OpenCensus API (i.e. using ServiceInfo.Name), I see that they're not taking namespace into account. I'll stick with this for now.

Reply:
Thanks for pointing that out. I was thinking to concatenate with a `-` instead of a `.` to avoid the ambiguity. Fine with me though to move forward with this as is.